Providers/Profile.swift

```diff
@@ -534,6 +535,8 @@ public class BrowserProfile: Profile {
 
         private var syncTimer: NSTimer? = nil
 
+        private var currentSync: Success? = nil
```

Comment:
This needs to be protected by a critical section, and we should use this instead of the lock for sync status. That is: the existence of a non-filled `currentSync` implies that a sync is in progress. A filled `Deferred` here, or this being `nil`, implies that no sync is in progress.